trigger/patch.go

```diff
@@ -189,6 +189,11 @@ func (t *patchTriggers) makeData(sub *event.Subscription) (*commonTemplateData,
 	projectName := t.patch.Project
 	if api.ProjectIdentifier != nil {
 		projectName = utility.FromStringPtr(api.ProjectIdentifier)
+	} else {
+		identifier, err := model.GetIdentifierForProject(*api.ProjectId)
```

Comment:
Assuming we're confident that `IncludeProjectIdentifier` is always set when needed for patch triggers, it seems like this wouldn't do anything, because [`api.ProjectId` is the same as the patch's project](https://github.com/evergreen-ci/evergreen/blob/24b4506727504198dd5cbbd47fb020c3e7e0a89a/rest/model/patch.go#L168), so this check seems equivalent to the if block above this. Are we certain that `t.patch.Project` is always set? If not, it seems like this would have the same issue as the old logic.

Reply:
Maybe I'll just add the logging and the other checks, and see what happens.

Reply:
Okay, I removed them. The theory that the cause of it is because the project identifier was set to an empty string feels like a long shot, but it's the only theory at the moment. If that doesn't fix it, I'll have the logging to help trace it.

Reply:
Is it possible that that `ProjectId` is nil or empty? It seems like [this check](https://github.com/evergreen-ci/evergreen/blob/1e57833ee9e7c59cca434dd398574c0b3e6d98d1/rest/model/patch.go#L115) would skip all this logic if `Project` is empty (i.e. `ProjectId` is empty).

Reply:
It doesn't seem likely because the patch in question (which is just one example) does have it populated in the db.